packages/backend-ai/package.json

```diff
@@ -0,0 +1,35 @@
+{
+  "name": "@aws-amplify/backend-ai",
```

Comment:
We were (still are?) struggling with vision around packaging "ai features".

Couple of thoughts:
1. The coming "ai features" seem to be independent islands of functionality. I.e. `defineAi`, `ai-construct` (singular) kind of abstraction don't seem to be correct. E.g. conversational bot is different set of components than generating images from description. These seem to be clusters of functionality under single "theme" (ai).
2. Because of point 1. If we were to follow `auth` and `data` we'll end up with multitude of packages like `ai-conversation-handler-construct`, `ai-image-generation-construct` etc. This seems a bit too aggressive and fragmented approach. Especially that we're introducing lower level public components like "conversation handler" (and there doesn't seem to be alternative for that).
3. The reason why `data` and `auth` constructs are separated is that there was hard requirement for them to be usable in raw CDK. On the other hand test I wrote seems to indicate that data + conversation construct could be used in raw CDK. (and a good signal to reconsider dumping everything to `backend-ai`).
4. While prototyping it felt odd that "construct package" would host 99% of the code now and "backend-ai" would be rather a facade. But, having that split eagerly might not be a bad idea to show next set of features a trajectory to follow.
5. The "conversation handler" construct is exposed publicly, so that both conversation transformer AND defineConversationHandler can see and use it. There didn't seem to be other need for construct to be public. But then `defineConversationHandler` translates to `new Construct` in raw CDK world. But we don't have strict requirement for raw CDK, so conversation construct being public is a necessity rather than desire right now.


I think we should take your suggestion, with small modification. I.e. do this:
1. Introduce a commonjs package called `ai-constructs` (plural!) . That will host any raw CDK components we need for ai "theme". (Today it's gonna be construct + lambda runtime components)
2. `backend-ai` to adapt raw CDK primitives to "Gen2 backend style". I.e. today it'll have `defineConversationHandler + re-exports of lambda runtime apis).

Additional supporting thought for such split is that data construct depending on ai-constructs seems to smell much better than data construct depending on backend-ai.

@Amplifiyer @atierian @josefaidt @dbanksdesign any objections to this ?

Reply:
Discussed offline. We'll create `ai-constructs` package to host CDK stuff.